pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1786,9 +1787,16 @@ func (vs *vSphereVMProvider) vmUpdateGetArgs(
 		return nil, err
 	}
 
-	resourcePolicy, err := GetVMSetResourcePolicy(vmCtx, vs.k8sClient)
-	if err != nil {
-		return nil, err
+	var resourcePolicy *vmopv1.VirtualMachineSetResourcePolicy
+	if vmCtx.VM.Annotations[pkg.ClusterModuleNameKey] != "" {
```

Comment:
Can we move `ClusterModuleNameKey` into `pkg/constants/constants.go` to avoid importing the `pkg` package here and also the webhook code below?